common/hexutil/json_test.go

```diff
@@ -57,7 +59,7 @@ func referenceBytes(s string) []byte {
 
 var unmarshalBytesTests = []unmarshalTest{
 	// invalid encoding
-	{input: "", wantErr: errNonString},
+	{input: "", wantErr: errors.New("unexpected end of JSON input")},
```

Comment:
Would be nice to put this error in its own variable. You're reusing it in quite a few places.